cpp/dolfinx/mesh/PartitioningNew.h

```diff
@@ -144,6 +146,31 @@ class PartitioningNew
   fetch_data(MPI_Comm comm, const std::vector<std::int64_t>& indices,
              const Eigen::Ref<const Eigen::Array<
                  double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>>& x);
+
+  /// Given an adjacency list with global, possibly non-contiguous, link
+  /// indices and a local adjacency list with contiguous link indices
+  /// starting from zero, compute a local-to-global map for the links.
+  /// Both adjacency lists must have the same shape.
+  /// @param[in] global Adjacency list with global link indices
+  /// @param[in] local Adjacency list with local, contiguous link
+  ///   indices
+  /// @return Map from local index to global index, which if applied to
+  /// the local adjacency list indices would yield the global adjacency
+  /// list
+  static std::vector<std::int64_t> compute_local_to_global_links(
+      const graph::AdjacencyList<std::int64_t>& global,
+      const graph::AdjacencyList<std::int32_t>& local);
+
+  /// Compute a local0-to-local1 map from two local-to-global maps with
```

Comment:
Is there a better name than `local_to_local`? The global indices of `local0` must be a subset of `local1`... maybe that can be stated in a comment or included in the function name?

Reply:
Let's revisit this one later.